models/articles.models.js

```diff
@@ -47,4 +47,32 @@ exports.fetchArticles = async (req, res, next) => {
 
     const result = await db.query(SQLquery)
     return result.rows
+}
+
+exports.addCommentsByArticleId = async (article_id, username, body) => {
+    if(!username || !body) {
+        return Promise.reject({ status: 400, msg: "Bad request: must include both username and body"})
+    }
+    
+    const checkArticle = await db.query(
+            `SELECT * FROM articles WHERE article_id = $1;`,
+            [article_id]
+        )
+    if (!checkArticle.rows.length) {
+            return Promise.reject({ status: 404, msg: `Article not found for article_id: ${article_id}` });
+        }
+    const checkAuthor = await db.query(
```

Comment:
This is good, but you don't need to check this. This is because article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value.
This means you don't need to check separately (comment these checks out and run the code to see what I mean - you have one case covered with 22P02 but there will be more error codes to be checked if you decide to refactor this)